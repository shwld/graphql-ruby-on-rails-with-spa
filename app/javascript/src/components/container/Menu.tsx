import React from 'react'
import { Link } from 'react-router-dom'

import { useCurrentUser } from '@/src/hooks/currentUser'

type Props = object

// type State = {}

export const Menu: React.FC<Props> = ({ children }) => {
  const { currentUser } = useCurrentUser()
  return (
    <>
      {currentUser && (
        <ul className="flex items-center justify-end p-6 bg-teal-500">
          <li className="mr-6">
            <Link to="/chat" className="text-white hover:text-yellow-800">
              {currentUser.email}
            </Link>
          </li>
          <li>
            <a href="/users/sign_out" data-method="delete">
              Sign out
            </a>
          </li>
        </ul>
      )}
      {!currentUser && (
        <ul className="flex items-center justify-end p-6 bg-teal-500">
          <li className="mr-6">
            <a href="/users/sign_in">Sign in</a>
          </li>
          <li>
            <a href="/users/sign_up">Sign up</a>
          </li>
        </ul>
      )}
      {children}
      {!currentUser && <div>Please sign in</div>}
    </>
  )
}
