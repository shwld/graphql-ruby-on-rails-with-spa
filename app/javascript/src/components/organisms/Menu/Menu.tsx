import React from 'react'
import { Link } from 'react-router-dom'

import { UserItemFragment } from '@/graphql/generated'
import { useCurrentUser } from '@/src/hooks/currentUser'

export const MenuTemplate: React.VFC<{
  currentUser: UserItemFragment
}> = ({ currentUser }) => {
  return (
    <>
      {currentUser && (
        <ul className="flex items-center justify-end p-6 bg-teal-500">
          <li className="mr-6">
            <Link to="/chat" className="text-white hover:text-yellow-800">
              {currentUser.email}
            </Link>
          </li>
          <li>
            <a href="/users/sign_out" data-method="delete">
              Sign out
            </a>
          </li>
        </ul>
      )}
      {!currentUser && (
        <ul className="flex items-center justify-end p-6 bg-teal-500">
          <li className="mr-6">
            <a href="/users/sign_in">Sign in</a>
          </li>
          <li>
            <a href="/users/sign_up">Sign up</a>
          </li>
        </ul>
      )}
      {!currentUser && <div>Please sign in</div>}
    </>
  )
}

export const Menu: React.VFC = () => {
  const { currentUser } = useCurrentUser()
  if (currentUser == null) return <></>

  return <MenuTemplate currentUser={currentUser} />
}
