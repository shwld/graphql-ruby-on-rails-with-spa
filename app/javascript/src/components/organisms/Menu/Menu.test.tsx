import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { create } from 'react-test-renderer'

import { MenuTemplate } from './Menu'

describe('Organisms/Menu', () => {
  it('Snap Shot', () => {
    const component = create(
      <MemoryRouter>
        <MenuTemplate currentUser={{ id: 'test', email: 'test@example.com' }} />
      </MemoryRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
