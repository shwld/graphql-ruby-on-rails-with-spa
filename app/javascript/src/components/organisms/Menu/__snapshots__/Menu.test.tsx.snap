// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Menu Snap Shot 1`] = `
<ul
  className="flex items-center justify-end p-6 bg-teal-500"
>
  <li
    className="mr-6"
  >
    <a
      className="text-white hover:text-yellow-800"
      href="/chat"
      onClick={[Function]}
    >
      test@example.com
    </a>
  </li>
  <li>
    <a
      data-method="delete"
      href="/users/sign_out"
    >
      Sign out
    </a>
  </li>
</ul>
`;
